### PR TITLE
Remove local dataset notion

### LINKED ISSUE #2539: Remove the notion of several local_datasets within a local_ministep
the local_dataset type was used as an identifier of a group of parameters to be used across time when running kalmann filter. As we are not running across time anymore (since the last 5< years, we can remove this and assume all parameters within a ministep are to be updated.

### PR DESCRIPTION
**Pull request by BjarneHerland**
**Issue**
Resolves #2539

**Approach**
Moving functionality from `local_dataset_type` to `local_ministep_type`, the remove the former.

PR currently has two commits: first to make `libres` compile and run ctests, second to fix some issues and make the Python-based tests work. There will be at third commit to remove all traces of `local_dataset_type` but I expect a number of comments and suggestions on the first two to handle first.

### CODE REVIEW

File: libres/lib/analysis/update.cpp [hunk 1]
```diff
@@ -283,38 +282,30 @@ load_parameters(enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
...
-    int matrix_start_size = 250000;
-    int active_ens_size = meas_data_get_active_ens_size(forecast);
-    matrix_type *A = matrix_alloc(matrix_start_size, active_ens_size);
-
-    serialize_info_type *serialize_info = serialize_info_alloc(
-        target_fs, //src_fs - we have already copied the parameters from the src_fs to the target_fs
-        target_fs, ensemble_config, iens_active_index, 0, last_step, A,
-        cpu_threads);
-
     std::unordered_map<std::string, matrix_type *> parameters;
```
Comment (ManInFez) [inline libres/lib/analysis/update.cpp:285]: This no longer needs to be an unordered map, as we are only creating one matrix here.

File: libres/lib/analysis/update.cpp [hunk 2]
```diff
@@ -406,14 +379,16 @@ load_row_scaling_parameters(enkf_fs_type *target_fs,
...
-                local_dataset_get_row_scaling(dataset, key.c_str());
+                ministep->get_row_scaling(key);
+
             matrix_shrink_header(A, row_scaling->size(), matrix_get_columns(A));
             row_scaling_list.push_back(
                 std::pair<matrix_type *, const row_scaling_type *>(
                     matrix_alloc_copy(A), row_scaling));
         }
-        parameters[std::string(dataset_name)] = row_scaling_list;
+        parameters[ministep->name] = row_scaling_list;
```
Comment (ManInFez) [inline libres/lib/analysis/update.cpp:389]: We only produce one "row_scaling_list" so we do not need to have the map of parameters to ministep names.

File: res/enkf/local_ministep.py [hunk 3]
```diff
@@ -16,42 +16,58 @@ class LocalMinistep(BaseCClass):
...
     def __init__(self, ministep_key):
         raise NotImplementedError("Class can not be instantiated directly!")
 
-    # Will used the data keys; and ignore observation keys.
-    def __getitem__(self, data_key):
-        if isinstance(data_key, int):
-            raise TypeError("Keys must be strings, not int!")
-        if data_key in self:
-            return self._get_local_data(data_key)
+    def initEnsembleConfig(self, config):
```
Comment (ManInFez) [inline res/enkf/local_ministep.py:44]: this is not an init function, but a set function.

**Review by ManInFez — COMMENTED** [review]
This looks very good! It would be good to remove the unordered maps used previously to hold parameters pr dataset, as they are no longer needed

File: res/enkf/local_ministep.py (hunk 3 above)
Comment (BjarneHerland) [reply res/enkf/local_ministep.py:44]: > this is not an init function, but a set function.

Yeah...  it's used exclusively during init, but I see your point...

**Comment by ManInFez** [comment]
Remember to rebase and clean up the commit messages

File: res/enkf/local_ministep.py [hunk 4]
```diff
@@ -16,42 +16,58 @@ class LocalMinistep(BaseCClass):
...
     def __init__(self, ministep_key):
         raise NotImplementedError("Class can not be instantiated directly!")
 
-    # Will used the data keys; and ignore observation keys.
-    def __getitem__(self, data_key):
-        if isinstance(data_key, int):
-            raise TypeError("Keys must be strings, not int!")
-        if data_key in self:
-            return self._get_local_data(data_key)
+    def setEnsembleConfig(self, config):
```
Comment (ManInFez) [inline res/enkf/local_ministep.py:44]: perhaps we should avoid camel-casing?

**Comment by BjarneHerland** [comment]
test ert please

**Comment by BjarneHerland** [comment]
test ert please

**Review by ManInFez — APPROVED** [review]

**Pull request merged**